util/fp_finder.go

```diff
@@ -0,0 +1,209 @@
+	}
+
+	// Filter words not in dictionary, remove duplicates, and sort alphabetically
+	filteredWords := t.filterContent(inputFile, dict, minSize)
+
+	if uniqEnabled {
+		filteredWords = t.removeDuplicates(filteredWords)
+	}
+
+	if sortEnabled {
+		sort.Slice(filteredWords, func(i, j int) bool {
+			return strings.ToLower(filteredWords[i]) < strings.ToLower(filteredWords[j])
+		})
+	}
+
+	for _, str := range filteredWords {
+		fmt.Println(str)
+	}
+
+	return nil
+}
+
+func (t *FpFinder) getDictionaryPath() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	crsToolchainDir := filepath.Join(homeDir, ".crs-toolchain")
+
+	// Create ~/.crs-toolchain folder if it doesn't exist
+	if _, err := os.Stat(crsToolchainDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(crsToolchainDir, 0755); err != nil {
+			return "", err
+		}
+	}
+
+	return filepath.Join(crsToolchainDir, dictionaryFileName), nil
+}
+
+func (t *FpFinder) downloadFile(filepath, url string) error {
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("bad status: %s", resp.Status)
+	}
+
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+func (t *FpFinder) loadDictionnary(path string, minSize int) (map[string]struct{}, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	content := make(map[string]struct{})
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		word := strings.TrimSpace(scanner.Text())
+		if len(word) >= minSize {
+			content[word] = struct{}{}
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	// Add words from the embedded extendedDictionnary
+	for _, word := range strings.Split(extendedDictionnary, "\n") {
+		word = strings.TrimSpace(word)
+		content[word] = struct{}{}
+	}
+
+	return content, nil
+}
+
+func (t *FpFinder) loadFileContent(path string) ([]string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	defer file.Close()
+
+	var content []string
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		word := strings.TrimSpace(scanner.Text())
+		content = append(content, word)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return content, nil
+}
+
+func (t *FpFinder) removeDuplicates(input []string) []string {
+	seen := make(map[string]struct{})
+	var result []string
+
+	for _, item := range input {
+		if _, exists := seen[item]; !exists {
+			seen[item] = struct{}{}
+			result = append(result, item)
+		}
+	}
+
+	return result
+}
+
+func (t *FpFinder) filterContent(inputFile []string, dict map[string]struct{}, minSize int) []string {
+	var filteredWords []string
+	for _, word := range inputFile {
+		if strings.HasPrefix(word, "#") {
```

Comment:
I usually filter with `\s*#` just to make sure that a space that has slipped in doesn't make a difference.